python/hawkey/subject-py.c

```diff
@@ -209,6 +210,49 @@ reldep_possibilities_real(_SubjectObject *self, PyObject *args, PyObject *kwds)
         csack, flags);
     return possibilitiesToPyObject(iter, sack);
 }
+static PyObject *
+get_best_solution(_SubjectObject *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *sack;
+    DnfSack *csack;
+    PyObject *form = NULL;
+    PyObject *icase = NULL;
+    PyObject *with_nevra = NULL;
+    PyObject *with_provides = NULL;
+    PyObject *with_filenames = NULL;
+    const char *kwlist[] = {"sack", "form", "icase", "with_nevra","with_provides","with_filenames",NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!|OO!O!O!O!", (char**) kwlist, &sack_Type, &sack, &form,
+        &PyBool_Type, &icase, &PyBool_Type, &with_nevra, &PyBool_Type, &with_provides, &PyBool_Type, &with_filenames)) {
+        return NULL;
+    }
+    HyForm *cforms = NULL;
+    if (form != NULL) {
+        cforms = fill_form(form);
+        if (cforms == NULL)
+            return NULL;
+    }
+    gboolean c_icase = icase!=NULL && PyObject_IsTrue(icase);
+    gboolean c_with_nevra = icase == NULL || PyObject_IsTrue(with_nevra);
+    gboolean c_with_provides = icase == NULL || PyObject_IsTrue(with_provides);
+    gboolean c_with_filenames = icase == NULL || PyObject_IsTrue(with_filenames);
+    csack = sackFromPyObject(sack);
+    HyNevra nevra = NULL;
+
+    HyQuery query = hy_subject_get_best_solution(self->pattern, csack, cforms, &nevra, c_icase, c_with_nevra, c_with_provides, c_with_filenames);
+
+    PyObject *q = queryToPyObject(query, sack);
+    PyObject *ret_dict = PyDict_New();
+    PyDict_SetItem(ret_dict, PyString_FromString("query"), q);
+    if (nevra != NULL) {
+        PyObject *n = nevraToPyObject(nevra);
```

Comment:
Dead code. nevra is allways NULL !

Reply:
Are you sure? The function hy_subject_get_best_solution gets pointer &nevra and the function can store there the nevra object.

Reply:
Sorry. You are right. I overlooked.